internal/controllers/reconciliation/controller.go

```diff
@@ -105,9 +105,12 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 
 	var prev *reconstitution.Resource
 	if comp.Status.PreviousSynthesis != nil {
-		synRef.UUID = comp.Status.PreviousSynthesis.UUID
-		prev, _ = c.resourceClient.Get(ctx, synRef, &req.Resource)
+		prevSynRef := reconstitution.NewSynthesisRef(comp)
```

Comment:
Avoid mutating the synthesis ref struct by just constructing a new one first here.